docs/reference/rest_api_names.md

```diff
@@ -0,0 +1,143 @@
+# Names
+
+Query and retrieve entries from the **names** vocabulary.
```

Comment:
Correct, it's for the creators/contributors autocomplete. 

I think the most relevant link in the docs is the [Names vocabulary customization](https://inveniordm.docs.cern.ch/customize/vocabularies/names/) (which also gives context on where this is used)... It feels a bit backwards though, since someone looking for "What is Names for?" would probably have a look at the Reference first, and then happen to stumble on a link to the Customization entry in the docs... 

Maybe I add the links now, and we discuss afterward how to organize things better both in terms of "going through the docs to install/customize" and "looking up things".